content/docs/command-reference/exp/gc.md

```diff
@@ -40,9 +40,10 @@ separately to delete it.
   the workspace (implies `-w`). Note that both options can be combined, for
   example using the `-aT` flag.
 
-- `--all-commits` - same as `-a` or `-T` above, but applies to _all_ Git commits
-  as well as the workspace (implies `-w`). This is mainly only needed when
-  clearing the experiments run queue.
+- `--all-commits` - same as `-a` or `-T` above, but applies to _all_
+  [reachable](https://git-scm.com/docs/gitglossary#def_reachable) Git commits as
+  well as the workspace (implies `-w`). This is mainly only needed when clearing
+  the experiments run queue.
```

Comment:
This is incorrect - `--all-commits` just includes all commits which are ancestors of any git tags or branches in the git repo. This is not the same things as "all reachable commits".

All reachable commits would include any git commit object which is reachable from any git reference at all, including things like `git stash` merge commits, `git notes`, and anything referenced in custom git refs (like DVC experiment commits - which are explicitly not included in `--all-commits`).

Reply:
Thanks for the answers and review @pmrowla ! No rush here, please enjoy your vacation 🌴

Reply:
Good to know. Fortunately the official definition of "reachable" is vague enough that it can mean what we want it to mean, I think (since we don't specify reachable from where).

![image](https://user-images.githubusercontent.com/1477535/112094391-c8fbd280-8b60-11eb-8417-5bab4dbc384a.png)
> https://git-scm.com/docs/gitglossary#def_reachable

> --all-commits just includes all commits which are ancestors of any git tags or branches in the git repo.

This makes me wonder though, is there a difference between `-A` and `-aT`? 🤔
_Unrelated to this PR so no rush_

> I don't think we should get into discussing concepts like "reachable" git commits

I think that saying "all commits" is definitely wrong though. Esp. now that we have checkpoints (and we mention that they are implemented as commits).

Reply:
Moved "reachable commit" stuff to #2300. PTAL when/if you can.